include/swift/Subsystems.h

```diff
@@ -249,18 +252,22 @@ namespace swift {
 
   /// Turn the given Swift module into either LLVM IR or native code
   /// and return the generated LLVM IR module.
+  /// If you set an outModuleHash, then you need to call performLLVM.
```

Comment:
This would be a pretty invasive change to the clients in REPL and Immediate mode. It is definitely the right way to design this and we can move to that soon, but this should be a NFC we can introduce later to keep this functionality change's change set small and targeted.